emu/src/cpu/arm/operations.rs

```diff
@@ -841,6 +836,264 @@ impl Arm7tdmi {
 
         self.flush_pipeline();
     }
+
+    pub fn multiply(&mut self, mul_variant: ArmModeMultiplyVariant, op_code: ArmModeOpcode) {
+        use ArmModeMultiplyVariant::*;
+
+        match mul_variant {
+            // Unsiged multiply (32-bit by 32-bit, bottom 32-bit result).
+            Mul => self.mul(op_code),
+            // Unsiged multiply-accumulate (32-bit by 32-bit, bottom 32-bit accumulate and result).
+            Mla => self.mla(op_code),
+        }
+    }
+
+    pub fn multiply_long(
+        &mut self,
+        mul_variant: ArmModeMultiplyLongVariant,
+        op_code: ArmModeOpcode,
+    ) {
+        use ArmModeMultiplyLongVariant::*;
+
+        match mul_variant {
+            // Unsigned long multiply (32-bit by 32-bit, 64-bit result)
+            Umull => self.umull(op_code),
+            // Unsigned long multiply-accumulate (32-bit by 32-bit, 64-bit accumulate and result)
+            Umlal => self.umlal(op_code),
+            // Signed long multiply (32-bit by 32-bit, 64-bit result)
+            Smull => self.smull(op_code),
+            // Signed multiply-accumulate (32-bit by 32-bit, 64-bit accumulate and result)
+            Smlal => self.smlal(op_code),
+        }
+    }
+
+    pub fn mul(&mut self, op_code: ArmModeOpcode) {
+        let set_condition_codes = op_code.get_bit(20);
+        assert!(0b1001u32 == op_code.get_bits(4..=7));
+        assert!(0u32 == op_code.get_bits(25..=27));
+
+        let rm_operand_register = op_code.get_bits(0..=3);
+        let rs_operand_register = op_code.get_bits(8..=11);
+        let rd_destination_register = op_code.get_bits(16..=19);
+
+        let rm_operand_value = self
+            .registers
+            .register_at(rm_operand_register.try_into().unwrap());
+        let rs_operand_value = self
+            .registers
+            .register_at(rs_operand_register.try_into().unwrap());
+
+        let result = u32::overflowing_mul(rm_operand_value, rs_operand_value);
+        self.registers
+            .set_register_at(rd_destination_register.try_into().unwrap(), result.0);
+
+        if set_condition_codes {
+            self.cpsr.set_carry_flag(result.1);
+            self.cpsr.set_overflow_flag(result.1);
```

Comment:
I must have missed this part, thanks :+1:

Reply:
I don't quite understand what is meant by `The C (Carry) flag is set to a meaningless`, what is it practically?

Reply:
What I understood is that the instruction does not guarantee to leave the `C` flag untouched, so the following instructions cannot use this value since it would be "undefined" in some way.

I guess that for the sake of the emulator we can leave it untouched